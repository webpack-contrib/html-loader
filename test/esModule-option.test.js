import {
  compile,
  execute,
  getCompiler,
  getErrors,
  getModuleSource,
  getWarnings,
  readAsset,
} from "./helpers";

describe("'esModule' option", () => {
  it("should use a CommonJS export by default", async () => {
    const compiler = getCompiler("simple.js");
    const stats = await compile(compiler);

    expect(getModuleSource("./simple.html", stats)).toMatchSnapshot("module");
    expect(
      execute(readAsset("main.bundle.js", compiler, stats))
    ).toMatchSnapshot("result");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should use a CommonJS export when the value is "false"', async () => {
    const compiler = getCompiler("simple.js", { esModule: false });
    const stats = await compile(compiler);

    expect(getModuleSource("./simple.html", stats)).toMatchSnapshot("module");
    expect(
      execute(readAsset("main.bundle.js", compiler, stats))
    ).toMatchSnapshot("result");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });

  it('should use an ES module export when the value is "true"', async () => {
    const compiler = getCompiler("simple.js", { esModule: true });
    const stats = await compile(compiler);

    expect(getModuleSource("./simple.html", stats)).toMatchSnapshot("module");
    expect(
      execute(readAsset("main.bundle.js", compiler, stats))
    ).toMatchSnapshot("result");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });
});
