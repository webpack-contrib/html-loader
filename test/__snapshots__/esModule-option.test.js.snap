// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'esModule' option should use a CommonJS export by default 1`] = `
"// Exports
module.exports = \\"<p>Hello world!</p>\\";"
`;

exports[`'esModule' option should use a CommonJS export when the value is "false" 1`] = `
"// Exports
module.exports = \\"<p>Hello world!</p>\\";"
`;

exports[`'esModule' option should use an ES module export when the value is "true" 1`] = `
"// Exports
export default \\"<p>Hello world!</p>\\";"
`;
